chronos/asyncfutures2.nim

```diff
@@ -398,6 +402,39 @@ proc `cancelCallback=`*[T](future: Future[T], cb: CallbackFunc) =
   ## This callback will be called immediately as ``future.cancel()`` invoked.
   future.cancelcb = cb
 
+{.push stackTrace: off.}
+proc futureContinue*[T](fut: Future[T]) {.gcsafe, raises: [Defect].} =
+  # Used internally by async transformation
+  try:
+    if not (fut.closure.finished()):
+      var next = fut.closure(fut)
+      while (not next.isNil()) and
+          next.finished():
+        next = fut.closure(fut)
+        if fut.closure.finished():
+          break
+      if next == nil:
+        if not (fut.finished()):
+          const
+            msg = "Async procedure (&" & "testInner" &
+                ") yielded `nil`, " &
+                "are you await\'ing a `nil` Future?"
+          raiseAssert msg
+      else:
+        next.addCallback(proc (arg: pointer) {.gcsafe, raises: [Defect].} =
+          futureContinue(fut))
+  except CancelledError:
+    fut.cancelAndSchedule()
+  except CatchableError as exc:
+    fut.fail(exc)
+  except Exception as exc:
+    if exc of Defect:
+      raise (ref Defect)(exc)
+    fut.fail((ref ValueError)(msg: exc.msg,
```

Comment:
This will never be reached with strictExceptions, but no need to remove it, afaik